plugin/evm/testutils/testutils_test.go

```diff
@@ -0,0 +1,81 @@
+package testutils
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestMustNotImport fails if a package in plugin/evm imports the testutils package.
+func TestMustNotImport(t *testing.T) {
+	var evmPackagePaths []string
+	filepath.WalkDir("../", func(path string, d fs.DirEntry, err error) error {
+		if !d.IsDir() {
+			return nil
+		} else if path == "../testutils" {
+			return nil
+		}
+		evmPackagePaths = append(evmPackagePaths, path)
+		return nil
+	})
+
+	for _, packagePath := range evmPackagePaths {
+		packageName := "github.com/ava-labs/subnet-evm/plugin/evm/" + strings.TrimPrefix(packagePath, "../")
+		imports, err := getPackageImports(packagePath)
+		require.NoError(t, err)
+		_, ok := imports["github.com/ava-labs/subnet-evm/plugin/evm/testutils"]
+		assert.Falsef(t, ok, "package %s imports testutils: testutils should only be used outside plugin/evm.", packageName)
+	}
+}
+
+func getPackageImports(packagePath string) (imports map[string]struct{}, err error) {
```

Comment:
There's an existing test in coreth that does something similar in case you wanted to adopt that style,
this is not exactly the same as here you walk plugin/evm to get the subdirs.

https://github.com/ava-labs/coreth/blob/master/plugin/evm/imports_test.go#L45

Reply:
Ha I know, I shamelessly copied and modified your code 😄 Thanks 💯